### PR TITLE
Add schema based condition

### PR DESCRIPTION
**Pull request by edgarmueller**
Adds a new 'SchemaBasedCondition' condition type

### CODE REVIEW

File: packages/core/src/util/runtime.ts [hunk 1]
```diff
@@ -25,48 +25,70 @@
...
+  (!_.has(uischema, 'rule.condition.expectedValue') && !_.has(uischema, 'rule.condition.schema'));
 
-export const evalVisibility = (uischema: UISchemaElement, data: any) => {
-  // TODO condition evaluation should be done somewhere else
-  if (isRuleDefined(uischema)) {
+const isLeafCondition = (condition: Condition): condition is LeafCondition =>
+  condition.type === 'LEAF';
 
+const isSchemaCondition = (condition: Condition): condition is SchemaBasedCondition =>
+  condition.type === undefined;
```
Comment (eneufeld) [inline packages/core/src/util/runtime.ts]: I would check for the schema attribute

File: packages/core/src/util/runtime.ts [hunk 2]
```diff
@@ -25,48 +25,70 @@
...
 };
 
-export const evalEnablement = (uischema: UISchemaElement, data: any) => {
-
-  if (isRuleDefined(uischema)) {
+export const evalVisibility = (uischema: UISchemaElement, data: any) => {
+  const fulfilled = isConditionFulfilled(uischema, data);
 
-    return true;
+  switch (uischema.rule.effect) {
```
Comment (eneufeld) [inline packages/core/src/util/runtime.ts]: can we unify the formatting for evalVisibility and evalEnablement

File: packages/core/src/util/runtime.ts (hunk 1 above)
Comment (edgarmueller) [reply packages/core/src/util/runtime.ts]: Makes sense, fixed

File: packages/core/src/util/runtime.ts (hunk 2 above)
Comment (edgarmueller) [reply packages/core/src/util/runtime.ts]: Fixed

**Pull request merged**